src/features/device-state/routes/state-patch-v3.ts

```diff
@@ -334,6 +322,33 @@ export const statePatchV3: RequestHandler = async (req, res) => {
 					});
 				}
 
+				if (
+					Object.keys(metricsBody).length > 0 &&
+					(await shouldUpdateMetrics(uuid))
+				) {
+					updateFns.push(async (resinApiTx) => {
+						const latestDeviceMetricsRecord = await resinApiTx.get({
+							resource: 'device_metrics_record',
+							id: { is_reported_by__device: device.id },
+						});
```

Comment:
This can be merged into the `fetchData` request rather than adding an extra one

Reply:
Thanks, great hint also explicitly loading with readOnly transcation. 

I've added it to the device request by expanding the `reports__device_metrics_record`